src/query/query_executor.c

```diff
@@ -9152,14 +9152,33 @@ qexec_open_scan (THREAD_ENTRY * thread_p, ACCESS_SPEC_TYPE * curr_spec, VAL_LIST
 
 	      if (curr_spec->num_parallel_threads > 1)
 		{
-		  if (!curr_spec->parts && !oid_is_system_class (&curr_spec->s.cls_node.cls_oid) && !mvcc_is_mvcc_disabled_class (&curr_spec->s.cls_node.cls_oid) && !mvcc_select_lock_needed && thread_p->private_heap_id != 0)	/* Only for User table */
+		  if (xasl->topn_items || XASL_IS_FLAGED (xasl, XASL_TO_BE_CACHED))
+		    {
+		      curr_spec->flags = (ACCESS_SPEC_FLAG) (curr_spec->flags & ~ACCESS_SPEC_FLAG_MERGED_LIST);
+		    }
+		  if (curr_spec->pruning_type != DB_PARTITIONED_CLASS && !oid_is_system_class (&curr_spec->s.cls_node.cls_oid) && !mvcc_is_mvcc_disabled_class (&curr_spec->s.cls_node.cls_oid) && !mvcc_select_lock_needed && thread_p->private_heap_id != 0)	/* Only for User table */
 		    {
 		      /* Why thread_p->private_heap_id != 0? 
 		       * Because, if it is 0, it means that the scan is not executed in main thread.
 		       * So, we can't use parallel heap scan.
 		       */
 		      scan_type = S_PARALLEL_HEAP_SCAN;
 		    }
+		  else
+		    {
+		      if (curr_spec->pruning_type == DB_PARTITIONED_CLASS
+			  && !oid_is_system_class (&curr_spec->s.cls_node.cls_oid)
+			  && !mvcc_is_mvcc_disabled_class (&curr_spec->s.cls_node.cls_oid) && !mvcc_select_lock_needed
+			  && thread_p->private_heap_id != 0)
+			{
+			  /* DB_PARTITION_CLASS will be parallel-heap-scanned */
+			}
+		      else
+			{
+			  curr_spec->flags =
+			    (ACCESS_SPEC_FLAG) (curr_spec->flags | ACCESS_SPEC_FLAG_NO_PARALLEL_HEAP_SCAN);
+			}
+		    }
```

Comment:
아래처럼 중복 비교를 줄일 수 있을 것 같습니다.
`DB_PARTITIONED_CLASS` 여부에 대한 것만 다릅니다.
혹시 else의 else인 경우가 발생할 수 있나요?아래 제안 코드에서는 else의 else 코드는 제거한 상태입니다.
```suggestion
		  if (!oid_is_system_class (&curr_spec->s.cls_node.cls_oid) && !mvcc_is_mvcc_disabled_class (&curr_spec->s.cls_node.cls_oid) && !mvcc_select_lock_needed && thread_p->private_heap_id != 0)	/* Only for User table */
		    {
		      if (curr_spec->pruning_type == DB_PARTITIONED_CLASS)
			{
			  /* DB_PARTITION_CLASS will be parallel-heap-scanned */
			}
		      else
			{
			  /* Why thread_p->private_heap_id != 0? 
			   * Because, if it is 0, it means that the scan is not executed in main thread.
			   * So, we can't use parallel heap scan.
			   */
			  scan_type = S_PARALLEL_HEAP_SCAN;
			}
		    }
```

Reply:
3가지 경우를 고려해야합니다.
1. parallel heap scan을 하지 못하는 경우 -> 이후 partition class 등을 위해 ACCESS_SPEC_FLAG_NO_PARALLEL_HEAP_SCAN을 추가해야합니다.
2.  parallel heap scan이 가능하지만 partitioned_class 인 경우 -> 이 테이블(wrapper)는 parallel heap scan을 하지 않지만, 차후에 partition_class는 parallel heap scan이 가능해야합니다.
3. parallel heap scan이 가능하고 partitioned_class 도 아닌 경우 -> parallel heap scan을 해야합니다.
현재는 1번에 해당하는 ACCESS_SPEC_FLAG_NO_PARALLEL_HEAP_SCAN을 넣어줘야 하는 경우는 여기서 어떻게 핸들링하면 될까요?

Reply:
확인했습니다. 설명 감사합니다.